Core/HTTPSTransientBloomFilterSpecification.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  HTTPSTransientBloomFilterSpecification.swift
+//  Core
+//
+//  Copyright © 2018 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public struct HTTPSTransientBloomFilterSpecification: Equatable {
```

Comment:
This is starting to seem overly complex with too many layers. 

Why not just dump Core Data for this and store the files on disk?  Using Codable pattern and optionals vars will help with migration issues.

If the whitelist is not likely to grow too big, drop Core Data for that too.

Reply:
We discussed this and decided that it made sense to stick with CoreData over files on disk for structured data for the migration benefits.